src/script/sign.cpp

```diff
@@ -395,7 +395,7 @@ class DummySignatureChecker final : public BaseSignatureChecker
 {
 public:
     DummySignatureChecker() {}
-    bool CheckSig(const std::vector<unsigned char>& scriptSig, const std::vector<unsigned char>& vchPubKey, const CScript& scriptCode, SigVersion sigversion) const override { return true; }
+    bool CheckSig(const std::vector<unsigned char>& scriptSig, const std::vector<unsigned char>& vchPubKey, const CScript& scriptCode, SigVersion sigversion, const bool require_secure_single) const override { return true; }
```

Comment:
Why not just pass `unsigned int flags` into `CheckSig`? It seems like there might be flags in the future we want to check for as well, so why not bite the bullet and just pass in the flags?

Reply:
I could go either way, but as is this avoids an internal dependency in `CheckSig` on `flags`' implementation.

Reply:
It is also used in sign.cpp so I'd like to avoid using `flags`